Reactions/ReactorCvodeJacobian.cpp

```diff
@@ -40,16 +40,54 @@ cJac(
       (SUNMatrix_cuSparse_NNZ(J) == ncells * NNZ));
 
     const auto ec = amrex::Gpu::ExecutionConfig(ncells);
-    amrex::launch_global<<<nbBlocks, nbThreads, ec.sharedMem, stream>>>(
-      [=] AMREX_GPU_DEVICE() noexcept {
-        for (int icell = blockDim.x * blockIdx.x + threadIdx.x,
-                 stride = blockDim.x * gridDim.x;
-             icell < ncells; icell += stride) {
-          fKernelComputeAJchem(
-            icell, NNZ, react_type, csr_row_count_d, csr_col_index_d, yvec_d,
-            Jdata);
-        }
-      });
+
+    if (nbThreads <= 32) {
+      amrex::launch_global<32><<<nbBlocks, nbThreads, ec.sharedMem, stream>>>(
```

Comment:
Should we get rid of the `ifs` here and use `CVODE_NB_THREADS`?

Reply:
We could do that. And then remove the user parameter and just make it a compile time thing?

Reply:
I'm thinking so.

Reply:
Ok I did that.